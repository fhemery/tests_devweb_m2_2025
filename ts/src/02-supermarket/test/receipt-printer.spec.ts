import { Receipt } from "../src/model/Receipt";
import { ReceiptPrinter } from "../src/ReceiptPrinter";
import { Product } from "../src/model/Product";
import { ProductUnit } from "../src/model/ProductUnit";

describe('ReceiptPrinter', () => {
    it('should print an empty receipt', () => {
        const receipt = new Receipt();
        const receiptPrinter = new ReceiptPrinter();
        const result = receiptPrinter.printReceipt(receipt);
        expect(result).toMatchSnapshot();
        
        // Other way (a bit trickier as soon as there will be)
        expect(result).toBe("\nTotal:                              0.00");
    });

    it('should print a receipt with one item', () => {
        const receipt = new Receipt();
        receipt.addProduct(new Product('apples', ProductUnit.Kilo), 2.5, 1.99, 2.5 * 1.99);
        const receiptPrinter = new ReceiptPrinter();
        const result = receiptPrinter.printReceipt(receipt);
        expect(result).toMatchSnapshot();
    });
})