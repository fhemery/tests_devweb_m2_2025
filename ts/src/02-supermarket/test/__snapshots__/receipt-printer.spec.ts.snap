// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiptPrinter should print a receipt with one item 1`] = `
"apples                              4.98
  1.99 * 2.500

Total:                              4.98"
`;

exports[`ReceiptPrinter should print an empty receipt 1`] = `
"
Total:                              0.00"
`;
